core/src/main/scala/org/apache/spark/sql/store/StoreUtils.scala

```diff
@@ -162,7 +162,11 @@ object StoreUtils extends Logging {
       val primaryKey = {
         v match {
           case PRIMARY_KEY => ""
-          case _ => s"$PRIMARY_KEY ($v, $SHADOW_COLUMN_NAME)"
+          //case _ => s"$PRIMARY_KEY ($v, $SHADOW_COLUMN_NAME)"
+          case _ => s"$PRIMARY_KEY ($SHADOW_COLUMN_NAME)" // Removing partition column for
+          // primary key. This has been done as we dont have support for CLOB in primary key in
+          // desrailized form. Once that's fixed we should uncomment the above line and comment
+          // this line
```

Comment:
We should keep it for cases other than LOB and complex types (CLOB, BLOB, ArrayType, MapType, StructType, UDT).

Also we need to find a way to carry through VARCHAR at Spark layer to here. Right now VARCHAR just gets through as a StringType which gets mapped to CLOB in GemXD. Clean way would have been to add a new DataType at our end, but then it could be used anywhere in Spark code and we would be in trouble. So we will need to have it as a private DataType which somehow can be used only in createTable, and that gets changed to a StringType once we have done the mapping.